twpca/utils.py

```diff
@@ -59,56 +59,10 @@ def stable_rank(matrix):
     svals_squared = np.linalg.svd(matrix, full_matrices=False, compute_uv=False) ** 2
     return svals_squared.sum() / svals_squared.max()
 
-
-def compute_lowrank_factors(data, n_components, fit_trial_factors, nonneg, last_idx, scale=1.0):
-    """Gets initial values for factor matrices by SVD on trial-averaged data
-
-    Args:
-        data: array-like
-        n_components: int
-        fit_trial_factors: bool, whether to compute trial factors
-        last_idx: nd-array, list of ints holding last index before trial end
-        scale: scale neuron and time factors by this amount, default 1.0
+def inverse_softplus(y):
+    """Inverse of the softplus function
     """
-
-    n_neurons = data.shape[-1]
-    if n_neurons == 1:
-        time_fctr = np.nanmean(data, axis=0)
-        neuron_fctr = np.atleast_2d([1.0])
-    else:
-        # do matrix decomposition on trial-averaged data matrix
-        DecompModel = NMF if nonneg else TruncatedSVD
-        model = DecompModel(n_components=n_components)
-        time_fctr = model.fit_transform(np.nanmean(data, axis=0))
-        neuron_fctr = np.transpose(model.components_)
-
-    # rescale factors to same length
-    s_time = np.linalg.norm(time_fctr, axis=0)
-    s_neuron = np.linalg.norm(neuron_fctr, axis=0)
-    s = np.sqrt(s_time * s_neuron)
-
-    time_fctr = (time_fctr * s / s_time).astype(np.float32)
-    neuron_fctr = (neuron_fctr * s / s_neuron).astype(np.float32)
-
-    # apply inverse softplus
-    if nonneg:
-        inv_softplus = lambda y: np.log(np.exp(y + 1e-5) - 1)
-        time_fctr = inv_softplus(time_fctr)
-        neuron_fctr = inv_softplus(neuron_fctr)
-
-    if not fit_trial_factors:
-        return None, time_fctr, neuron_fctr
-
-    # If trial factors also need to be initialized, do a single step of alternating
-    # least squares (see Kolda & Bader, 2009) for CP tensor decomposition.
-    else:
-        Bpinv = np.linalg.pinv(neuron_fctr)
-        trial_fctr = np.empty((data.shape[0], n_components), dtype=np.float32)
-        for k, trial in enumerate(data):
-            t = last_idx[k] # last index before NaN
-            trial_fctr[k] = np.diag(np.linalg.pinv(time_fctr[:t]).dot(trial[:t]).dot(Bpinv.T))
-        return trial_fctr, time_fctr, neuron_fctr
-
+    return np.log(np.exp(y + 1e-5) - 1)
```

Comment:
could you make this smaller, and maybe put it in the log, like:
`np.log(np.exp(y) - 1 + 1e-9)`